iquip/apps/builder.py

```diff
@@ -365,22 +355,146 @@ def get_state(self, argInfo: Dict[str, Any]) -> Dict[str, Any]:
         procdesc = {
-            "unit": argInfo["unit"],
-            "scale": argInfo["scale"],
-            "global_step": argInfo["global_step"],
-            "global_min": argInfo["global_min"],
-            "global_max": argInfo["global_max"],
-            "ndecimals": argInfo["ndecimals"]
+            "unit": self.argInfo["unit"],
+            "scale": self.argInfo["scale"],
+            "global_step": self.argInfo["global_step"],
+            "global_min": self.argInfo["global_min"],
+            "global_max": self.argInfo["global_max"],
+            "ndecimals": self.argInfo["ndecimals"]
         }
         return procdesc
 
+    def value(self) -> Dict[str, Any]:
+        """Overridden.
+        
+        Returns the dictionary of the selected scannable arguments.
+        """
+        selected = self.state["selected"]
+        return self.state[selected]
+
+
+class _RangeScan(QWidget):
+    """A widget for range scan in _ScanEntry.
+
+    Attributes:
+        scale: The scale factor that is multiplied to the number value.
+        state: Each key and its value as follows.
+          start: The start point for the RangeScan sequence.
+          stop: The end point for the RangeScan sequence.
+          npoints: The number of points in the RangeScan squence.
+          randomize: A boolean value that decides whether to shuffle the RangeScan sequence.
+        start: QDoubleSpinBox for start arguments inside state.
+        stop: QDoubleSpinBox for stop arguments inside state.
+        npoints: QSpinBox for npoints arguments inside state.
+        randomize: QCheckBox for randomize arguments inside state.
+        layout: The layout of _RangeScan widget.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ):
+        """Extended.
+
+        Args: See the docstring of Attributes for more detail.
+            procdesc: Each key and its value are as follows.
+              unit: The unit of the number value.
+              scale: See scale docstring at class Attributes.
+              global_step: The step between values changed by the up and down button.
+              global_min: The minimum value. (default=float("-inf"))
+              global_max: The maximum value. (default=float("inf"))
+                If min > max, then they are swapped.
+              ndecimals: The number of displayed decimals.
+            state: See the state docstring at class Attributes.
+        """
+        super().__init__(parent=parent)
+        self.scale = procdesc["scale"]
+        self.state = state
+        self.layout = QGridLayout(self)
+        self.start = QDoubleSpinBox(self)
+        self.apply_properties(self.start, procdesc)
+        self.start.setValue(state["start"] / self.scale)
+        self.npoints = QSpinBox(self)
+        self.npoints.setMinimum(1)
+        self.npoints.setMaximum((1 << 31) - 1)
```

Comment:
This is a question! Is there any reason to set the maximum value to this?

Reply:
It was at original artiq code. I don't know exact number that is needed for experiments.